pkg/dogstatsd/server.go

```diff
@@ -446,12 +454,17 @@ func (s *Server) parseMetricMessage(parser *parser, message []byte, originTagsFu
 			sample.tags = append(sample.tags, mapResult.Tags...)
 		}
 	}
-	metricSample := enrichMetricSample(sample, s.metricPrefix, s.metricPrefixBlacklist, s.defaultHostname, originTagsFunc, s.entityIDPrecedenceEnabled, s.ServerlessMode)
-	metricSample.Tags = append(metricSample.Tags, s.extraTags...)
+	metricSamples = enrichMetricSample(metricSamples, sample, s.metricPrefix, s.metricPrefixBlacklist, s.defaultHostname, originTagsFunc, s.entityIDPrecedenceEnabled, s.ServerlessMode)
 
-	dogstatsdMetricPackets.Add(1)
-	tlmProcessed.IncWithTags(tlmProcessedOkTags)
-	return metricSample, nil
+	if len(sample.values) > 0 {
+		s.sharedFloat64List.put(sample.values)
+	}
+	for idx := range metricSamples {
+		metricSamples[idx].Tags = append(metricSamples[idx].Tags, s.extraTags...)
```

Comment:
Not for this PR just food for thoughts: this tag slices could probably be shared between this chunk of metricSamples, which could help with memory/gc & some cpu cycles (but not with code readability though 😅)

Reply:
Yes ! Good catch, let's do it now.